skrifa/src/meta/metrics.rs

```diff
@@ -0,0 +1,434 @@
+//! Global font and glyph specific metrics.
+
+use read_fonts::{
+    tables::{glyf::Glyf, hmtx::LongMetric, hvar::Hvar, loca::Loca},
+    types::{BigEndian, GlyphId},
+    TableProvider,
+};
+
+use crate::NormalizedCoord;
+
+/// Metrics for a text decoration.
```

Comment:
This would be a magnificent place to explain what a decoration *is*

Reply:
Also ... are there any invariants? What would a negative or 0 thickness mean?

Reply:
Looking ahead I think a "decoration" is always a line, presumably a line drawn on top of the normal content?

Reply:
Yes, decoration is an underline or strike through. I borrowed the terminology from css which I think would be familiar to most users? We could call this Line but to me that implies things like ascent, descent and leading in this context. 

There are no invariants— it simply returns what the font specifies. Both thickness values are signed in the spec so it seems reasonable to return absolute value here. Otherwise it should be up to the rendering implementation to decide what values make sense.

Reply:
Looking a bit deeper into this, FreeType does [weird stuff](https://github.com/freetype/freetype/blob/5c37b6406258ec0d7ab64b8619c5ea2c19e3c69a/src/sfnt/sfobjs.c#L1390) by shifting the position to be in the _middle_ of the stroke which doesn't seem to be useful.

Skia summarily [undoes this](https://github.com/google/skia/blob/04e32a9e56411e7fcdd894255323629ee6e7b103/src/ports/SkFontHost_FreeType.cpp#L1599) and also negates the offsets (converting from Y-up to Y-down).

The thickness values aren't touched at all.

I'd suggest we keep the current code as is, returning the raw values contained in the font.